test/cuNDArray_blas_test.cpp

```diff
@@ -7,150 +7,235 @@
 using namespace Gadgetron;
 using testing::Types;
 
-template <typename T> class cuNDArray_blas_Real : public ::testing::Test 
-{
-protected:
-  virtual void SetUp() {
-    size_t vdims[] = {37, 49, 23, 19}; //Using prime numbers for setup because they are messy
-    dims = std::vector<size_t>(vdims,vdims+sizeof(vdims)/sizeof(size_t));
-    Array = cuNDArray<T>(&dims);
-    Array2 = cuNDArray<T>(&dims);
-  }
-  std::vector<size_t> dims;
-  cuNDArray<T> Array;
-  cuNDArray<T> Array2;
+template <typename T> class cuNDArray_blas_Real : public ::testing::Test {
+  protected:
+    virtual void SetUp() {
+        size_t vdims[] = {37, 49, 23, 19}; // Using prime numbers for setup because they are messy
+
+        dims = std::vector<size_t>(vdims, vdims + sizeof(vdims) / sizeof(size_t));
+
+        // test updated code
+        size_t vdims_large[] = {size_t(INT_MAX) + size_t(10000)}; // Using a large vector to test updated code
+        dims_large = std::vector<size_t>(vdims_large, vdims_large + sizeof(vdims_large) / sizeof(size_t));
+
+        Array = cuNDArray<T>(&dims);
+        Array2 = cuNDArray<T>(&dims);
+        Array3 = cuNDArray<T>(&dims_large);
```

Comment:
I actually don't think you need two sets of arrays here. If you (as discussed) implement a max batch size in the various functions (default to `INT_MAX`), you can simply change the batch size (even parameterize that in the tests) to make sure you get the right result. That would also allow you to make sure you pick odd and even sizes of those (to make sure your rounding match is right in the batching).